PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/test/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/ConnectionStringCheckTest.java

```diff
@@ -0,0 +1,135 @@
+package com.microsoft.azure.toolkit.intellij.azure.sdk.buildtool;
+
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.psi.JavaElementVisitor;
+import com.intellij.psi.PsiClass;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiElementVisitor;
+
+import com.intellij.psi.PsiMethod;
+import com.intellij.psi.PsiMethodCallExpression;
+import com.intellij.psi.PsiReferenceExpression;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+
+/**
+ * This class tests the ConnectionStringCheck class.
+ * It tests the buildVisitor method and the visitElement method.
+ * It tests the check for the usage of connection strings API to create clients in the Azure SDK for Java.
+ *
+ * This
+ */
+public class ConnectionStringCheckTest {
+
+    @Mock
+    private ProblemsHolder mockHolder;
+
+    @Mock
+    private PsiElement mockElement;
+
+    @Mock
+    private PsiElementVisitor mockVisitor;
+
+
+    @BeforeEach
+    public void setup() {
+        mockHolder = mock(ProblemsHolder.class);
+        mockElement = mock(PsiElement.class);
+        mockVisitor = createVisitor();
+    }
+
+    PsiElementVisitor createVisitor() {
+        boolean isOnTheFly = true;
+        ConnectionStringCheck check = new ConnectionStringCheck();
+        return check.buildVisitor(mockHolder, isOnTheFly);
+    }
+
+    /**
+     * This test checks the buildVisitor method.
+     * It checks if the visitor is created and if it is an instance of JavaElementVisitor.
+     * It also verifies that a warning is raised when a connection string being used to create a client is detected.
+     *
+     * This is an example of a passing test.
+     * 1. ServiceBusSenderClient senderClient2 = new ServiceBusClientBuilder().connectionString(connectionString2).sender().queueName("myQueueName2").buildClient();
+     * 2. BlobServiceClient blobServiceClient2 = new BlobServiceClientBuilder().connectionString(connectionString2).buildClient();
+     */
+    @Test
+    void testConnectionStringCheck() {
+
+        assertVisitor(mockVisitor);
+        String METHOD_TO_CHECK = "connectionString";
```

Comment:
How are we testing that the `connectionString` method call is happening only in Azure SDK clients context?

Reply:
the stubbing in line 97 returns CLASS_NAME = "com.azure"; also there's a test in line 118 that tests for use of different clients, with 0 invocations of registered problem.